Mathlib/Algebra/Group/Defs.lean

```diff
@@ -620,11 +620,139 @@ the file `Algebra.GroupPower.Basic`. For now, we only register the most basic pr
 need right away.
 -/
 
+/--
+An abbreviation for `npowRec` with the typeclass hypotheses matching `npowBinRec`,
+so that we can use `@[csimp]` to replace it with an implementation by repeated squaring in compiled
+code.
+-/
+@[to_additive (attr := nolint unusedArguments)
+"An abbreviation for `nsmulRec` with the typeclass hypotheses matching `nsmulBinRec`,
+so that we can use `@[csimp]` to replace it with an implementation by repeated doubling in compiled
+code."]
+abbrev npowRec' {M : Type*} [Semigroup M] [One M]
+    (_mul_one : ∀ m : M, m * 1 = m) (_one_mul : ∀ m : M, 1 * m = m) (k : ℕ) (m : M) : M :=
```

Comment:
Nit: you could instead use
```suggestion
abbrev npowRec' {M : Type*} [MulOneClass M]
    (_mul_assoc : sorry) (k : ℕ) (m : M) : M :=
```
to reduce the number of arguments here (and for the others too)

Reply:
This no longer seems relevant so I am resolving. Please let me know if I am wrong.